src/collector.rs

```diff
@@ -188,11 +207,15 @@ async fn mk_emitter(
     while let Some(metrics) = emit_receiver.recv().await {
         stream::iter(metrics_chunks(&metrics))
             .for_each(|metric_data| async move {
-                let send_fut = cloudwatch_client.put_metric_data(PutMetricDataInput {
-                    metric_data: metric_data.to_owned(),
-                    namespace: cloudwatch_namespace.clone(),
+                let send_fut = retry_on_throttled(|| async {
+                    cloudwatch_client
+                        .put_metric_data(PutMetricDataInput {
+                            metric_data: metric_data.to_owned(),
+                            namespace: cloudwatch_namespace.clone(),
+                        })
+                        .await
                 });
-                match tokio::time::timeout(SEND_TIMEOUT, send_fut).await {
+                match send_fut.await {
```

Comment:
How will waiting for a maximum of 6 seconds affect our channel `emit_receiver`?

Reply:
I suppose there is more of a risk that we start falling behind, I don't know what a good sleep time would be though so :shrug:

Reply:
https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/cloudwatch_limits.html

>  PutMetricData can handle 150 transactions per second (TPS), which is the maximum number of operation requests you can make per second without being throttled.

Reply:
I suppose this is per account, so one instance of metrics cloudwatch can be rate limited by another one, so it can't really be determined. 150 seems fairly low though. Not sure retry is the way to go since it means we fall behind, we might want to request a quota increase instead?

Reply:
I checked the count for putmetricdata and we used to send A LOT more requests so I think we should still be fine, this is more likely just a consequence of trying to send metrics to quickly succession.